bw2calc/lca.py

```diff
@@ -517,39 +517,47 @@ def redo_lcia(self, demand=None):
         #     extra={"demand": wrap_functional_unit(demand or self.demand)},
         # )
 
-    # def to_dataframe(self, cutoff=200):
-    #     """Return all nonzero elements of characterized inventory as Pandas dataframe"""
-    #     assert mapping, "This method doesn't work with independent LCAs"
-    #     assert (
-    #         pandas
-    #     ), "This method requires the `pandas` (http://pandas.pydata.org/) library"
-    #     assert hasattr(
-    #         self, "characterized_inventory"
-    #     ), "Must do LCIA calculation first"
-
-    #     from bw2data import get_activity
-
-    #     coo = self.characterized_inventory.tocoo()
-    #     stacked = np.vstack([np.abs(coo.data), coo.row, coo.col, coo.data])
-    #     stacked.sort()
-    #     rev_activity, _, rev_bio = self.reverse_dict()
-    #     length = stacked.shape[1]
-
-    #     data = []
-    #     for x in range(min(cutoff, length)):
-    #         if stacked[3, length - x - 1] == 0.0:
-    #             continue
-    #         activity = get_activity(rev_activity[stacked[2, length - x - 1]])
-    #         flow = get_activity(rev_bio[stacked[1, length - x - 1]])
-    #         data.append(
-    #             (
-    #                 activity["name"],
-    #                 flow["name"],
-    #                 activity.get("location"),
-    #                 stacked[3, length - x - 1],
-    #             )
-    #         )
-    #     return pandas.DataFrame(data, columns=["Activity", "Flow", "Region", "Amount"])
+    def to_dataframe(self, cutoff=200):
+        """Return all nonzero elements of characterized inventory as Pandas dataframe"""
+        # assert mapping, "This method doesn't work with independent LCAs"
+        assert (
+            pandas
+        ), "This method requires the `pandas` (http://pandas.pydata.org/) library"
+        assert hasattr(
+            self, "characterized_inventory"
+        ), "Must do LCIA calculation first"
+
+        from bw2data import get_activity
```

Comment:
`bw2data` is an optional dependency of `bw2calc`. Please see the way this is handled in `MultiLCA` (regarding `bw2data.calculation_setups`). 
(This is the opposite of the pandas requirement above)

Reply:
For example

```python3
try:
    from bw2data import get_activity
except ImportError:
    get_activity = None
...
if get_activity is None:
            raise ImportError("`bw2data` is required for this functionality")
```

Reply:
Actually, maybe we can implement in a way that it works even without `get_activity`.
`get_activity` should be used if we have it available, and hence the returned DataFrame would have a name column, but if we don't have it, we could use whatever the `self.dicts.biosphere.reversed` provide.
I'm not sure though if these (`self.dicts.X`) are systematically available.